### DISCUSSION TITLE
HEARTBEAT_DEAD vs MACHINE_HEARTBEAT_DEAD

### DISCUSSION THREAD

**Discussion by nCubed** [keygen-sh/community#100] [Issue]
In reviewing the docs for the [policy heartbeatResurrectionStrategy](https://keygen.sh/docs/api/policies/#policies-object-attrs-heartbeatResurrectionStrategy), the docs state dead machines will return a `HEARTBEAT_DEAD` error upon validation. I noticed the error I received today was for `MACHINE_HEARTBEAT_DEAD`. Is there a difference between these two errors?

Of note, I actually received the `MACHINE_HEARTBEAT_DEAD` error when sending a heartbeat from a dead machine (the machine had not been properly deactivated).

**Comment by ezekg** [comment]
One is a validation code, and the other is an error code, respectively. The validation code is not an error, it's part of the validation result. The error is because you can't ping a dead machine, unless the policy allows resurrections.
